python/cudf/cudf/utils/dtypes.py

```diff
@@ -144,16 +144,11 @@ def numeric_normalize_types(*args):
 
 
 def is_numerical_dtype(obj):
```

Comment:
Not sure if we want to match Pandas, but https://pandas.pydata.org/docs/reference/api/pandas.api.types.is_numeric_dtype.html shows that `obj` could be an array, and we handle the same in our `is_string_dtype` for example.

Reply:
We aren't relying on this right now, so can we make it a potential TODO if required later?

Reply:
Enabling a fastpath for when we know we're dealing with dtype objects would be ideal.